server/plugin.go

```diff
@@ -98,3 +96,15 @@ func (p *Plugin) MessageWillBePosted(c *plugin.Context, post *model.Post) (*mode
 
 	return post, ""
 }
+
+// MessageWillBePosted is invoked when a message is posted by a user before it is committed
+// to the database.
+func (p *Plugin) MessageWillBePosted(c *plugin.Context, post *model.Post) (*model.Post, string) {
+	return p.processPost(c, post)
+}
+
+// MessageWillBeUpdated is invoked when a message is updated by a user before it is committed
+// to the database.
+func (p *Plugin) MessageWillBeUpdated(c *plugin.Context, post *model.Post, _ *model.Post) (*model.Post, string) {
```

Comment:
This can break some existing use cases, but I think it the expected behavior for the most cases.